services/s3control/src/main/java/software/amazon/awssdk/services/s3control/S3ControlConfiguration.java

```diff
@@ -0,0 +1,151 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3control;
+
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.NotThreadSafe;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.core.ServiceConfiguration;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * S3 Control specific configuration allowing customers to enabled FIPS or
+ * dualstack.
+ */
+@SdkPublicApi
+@Immutable
+@ThreadSafe
+public final class S3ControlConfiguration implements ServiceConfiguration,
+                                                     ToCopyableBuilder<S3ControlConfiguration.Builder, S3ControlConfiguration> {
+    /**
+     * S3 FIPS mode is by default not enabled
+     */
+    private static final boolean DEFAULT_FIPS_MODE_ENABLED = false;
+
+    /**
+     * S3 Dualstack endpoint is by default not enabled
+     */
+    private static final boolean DEFAULT_DUALSTACK_ENABLED = false;
+
+    private final boolean fipsModeEnabled;
+    private final boolean dualstackEnabled;
+
+    private S3ControlConfiguration(DefaultS3ServiceConfigurationBuilder builder) {
+        this.dualstackEnabled = resolveBoolean(builder.dualstackEnabled, DEFAULT_DUALSTACK_ENABLED);
+        this.fipsModeEnabled = resolveBoolean(builder.fipsModeEnabled, DEFAULT_FIPS_MODE_ENABLED);
+    }
+
+    /**
+     * Create a {@link Builder}, used to create a {@link S3ControlConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultS3ServiceConfigurationBuilder();
+    }
+
+    /**
+     * <p>
+     * Returns whether the client has enabled fips mode for accessing S3 Control.
+     *
+     * @return True if client will use FIPS mode.
+     */
+    public boolean fipsModeEnabled() {
+        return fipsModeEnabled;
+    }
+
+    /**
+     * <p>
+     * Returns whether the client is configured to use dualstack mode for
+     * accessing S3. If you want to use IPv6 when accessing S3, dualstack
+     * must be enabled.
+     * </p>
+     *
+     * <p>
+     * Dualstack endpoints are disabled by default.
+     * </p>
+     *
+     * @return True if the client will use the dualstack endpoints
+     */
+    public boolean dualstackEnabled() {
+        return dualstackEnabled;
+    }
+
+    private boolean resolveBoolean(Boolean customerSuppliedValue, boolean defaultValue) {
```

Comment:
Consider renaming customerSuppliedValue to builderSuppliedValue or even just suppliedValue.